public/main.js

```diff
@@ -70,34 +71,41 @@ function reloadTodoList() {
         todoListPlaceholder.style.display = "none";
         incompleteTodoCount = 0;
         var i = 0;
+        var promises = [];
         todosLocal.forEach(function(todo) {
-            var listItem = document.createElement("li");
-            listItem.textContent = todo.title;
+            promises.push(new Promise(function(resolve, reject) {
```

Comment:
Perhaps at this point we want to start pulling some functions out?
We are currently within an anonymous function within a constructor within an anonymous function (in a loop) within a function.
I would recommend putting the following logic of the page manipulation into a separate named function.